couler/core/syntax/loop.py

```diff
@@ -72,18 +82,44 @@ def map(function, input_list):
 
     inner_step.arguments = {"parameters": parameters}
 
+    # the following part of the code
+    # Adds values to parameters (with items) while it goes
+    # through the *arguments-variable with two loops.
+    # inner loop: arguments[z][0], arguments[z][0]...
+    # Outer loop: arguments[0][y], arguments[0][y]...
+    # With two lists in *arguments
+    # result would be:
+    #  1. pair of items for the function:
+    # arguments[0][0], arguments[1][0]
+    # 2. pair of items for the function:
+    #  arguments[0][1], arguments[1][1]
+    # and so on...
     with_items = []
-    for para_values in input_list:
+
+    # loops through the length of the lists inside *arguments
+    y = 0
```

Comment:
Please give more meaningful names to your indices.
Also, the comment text should reflect what `len(arguments[0])`  mean - the number of calls to be made to  `function`.

So `y` should be renamed, say, to `ind_of_func_call` or something along these lines.